src/os/bluestore/BlueStore.cc

```diff
@@ -1640,6 +1640,31 @@ int BlueStore::_setup_block_symlink_or_file(
 	  VOID_TEMP_FAILURE_RETRY(::close(fd));
 	  return r;
 	}
+#ifdef HAVE_POSIX_FALLOCATE
+	r = ::posix_fallocate(fd, 0, size);
+	if (r < 0) {
+	  r = -errno;
+	  derr << __func__ << " failed to prefallocate " << name << " file to "
+	       << size << ": " << cpp_strerror(r) << dendl;
+	  VOID_TEMP_FAILURE_RETRY(::close(fd));
+	  return r;
+	}
+#else
+	char data[1024*128];
```

Comment:
Sorry to reply so later. The code already use ftruncate.  But this don't allocate space. So like FileJournal, avoid later no space use and it should allocate space.